spec/unit/application_spec.rb

```diff
@@ -142,19 +142,28 @@
 
     it "should yield a new namespace" do
       application.namespace :new_namespace do |ns|
-        expect(ns.name).to eq :new_namespace
+        ActiveSupport::Deprecation.silence do
+          expect(ns.name).to eq :new_namespace
+          expect(ns.name_path).to eq [:admin, :new_namespace]
+        end
       end
     end
 
     it "should return an instantiated namespace" do
       admin = application.namespace :admin
-      expect(admin).to eq application.namespaces[:admin]
+      expect(admin).to eq application.namespaces[[:admin]]
```

Comment:
It would be nice to have
```
class Namespace::Store
  def [](key)
    @namespaces[Array(key)]
  end
end
```
Then all three cases in https://github.com/activeadmin/activeadmin/pull/5185#issuecomment-343724223 (A) would work.